vulture_os/toolkit/api_parser/crowdstrike/crowdstrike.py

```diff
@@ -160,146 +152,170 @@ def unionDict(self, dictBase, dictToAdd):
                 finalDict[k] = dictToAdd[k]
         return finalDict
 
-    def execute_query(self, method, url, query={}, timeout=10):
-        # can set a custom limit of entry we want to retrieve
-        customLimit = int(query.get('limit', -1))
-
-        jsonResp = self.__execute_query(method, url, query, timeout=timeout)
-        totalToRetrieve = jsonResp.get('meta', {}).get('pagination', {}).get('total', 0)
-
-        while(totalToRetrieve > 0 and totalToRetrieve != len(jsonResp.get('resources', []))):
-            # we retrieved enough data
-            if(customLimit > 0 and customLimit <= len(jsonResp['resources'])):
-                break
-            query['offset'] = len(jsonResp['resources'])
-            jsonAdditionalResp = self.__execute_query(method, url, query, timeout=timeout)
-            self.update_lock()
-            jsonResp = self.unionDict(jsonResp, jsonAdditionalResp)
-            #jsonResp += [jsonAdditionalResp]
-        return jsonResp
-
-    def getSummary(self):
-        nbSensor = self.getSensorsTotal()
-        version, updated = self.getApplianceVersion()
-        return nbSensor, version, updated
-
-    def getApplianceVersion(self):
-        return 0, 0
-
-    def getSensorsTotal(self):
-        device_url = f"{self.api_host}/{self.DEVICE_URI}"
-        ret = self.execute_query('GET', device_url, {'limit': 1})
-        return int(ret['meta']['pagination']['total'])
-
-    def getAlerts(self, since, to):
-        '''
-        we retrieve raw incidents and detections
-        '''
-        logger.info(f"[{__parser__}][getAlerts]: From {since} until {to}",  extra={'frontend': str(self.frontend)})
-
-        finalRawAlerts = []
-        # first retrieve the detection raw ids
-        alert_url = f"{self.api_host}/{self.DETECTION_URI}"
-        payload = {
-            "filter": f"last_behavior:>'{since}'+last_behavior:<='{to}'",
-            "sort": "last_behavior|desc"
-        }
-        ret = self.execute_query("GET", alert_url, payload)
-        ids = ret['resources']
-        if(len(ids) > 0):
-            # retrieve the content of detection selected
-            alert_url = f"{self.api_host}/{self.DETECTION_DETAILS_URI}"
-            payload = {"ids": ids}
-            ret = self.execute_query("POST", alert_url, payload)
-
-            alerts = ret['resources']
-            for alert in alerts:
-                finalRawAlerts += [alert]
 
-        if self.request_incidents:
-            # then retrieve the incident raw ids
-            alert_url = f"{self.api_host}/{self.INCIDENT_URI}"
-            payload = {
-                "filter": f"start:>'{since}'+start:<='{to}'",
-                "sort": "end|desc"
-            }
+    def get_ids(self, url: str, query: dict = {}, timeout: int = 10) -> list[list]:
```

Comment:
shouldn't the URL be static? seems like it's always the same one for that method